### PR TITLE
Support time varying gains

### PR DESCRIPTION
**Pull request by r-pascua**
This PR ports over and extends the tools developed for making time varying gains as used in validation tests 2.1 and 4.0.

### CODE REVIEW

File: hera_sim/tests/test_sigchain.py [hunk 1]
```diff
@@ -182,5 +185,339 @@ def test_cross_coupling_xtalk(self):
         )
 
 
+class TestTimeVariation(unittest.TestCase):
```
Comment (steven-murray) [inline hera_sim/tests/test_sigchain.py:188]: Don't use unittest stuff! Use `pytest` fixtures / parameterizations

Comment (r-pascua) [reply hera_sim/tests/test_sigchain.py:188]: Gotcha. I'll rewrite the new tests to use `pytest` only, and make the tests more atomic.

Comment (r-pascua) [reply hera_sim/tests/test_sigchain.py]: All of the tests for this module (including the old ones) have been updated to use `pytest` only. Some further consolidation can be done, and some tests may be able to be compactified by exploiting parameterization, but I'd like to leave that for the PR where I convert *all* of the tests to `pytest` only.

Comment (steven-murray) [reply hera_sim/tests/test_sigchain.py]: Excellent!

**Pull request merged**